desktop/core/src/desktop/api2_tests.py

```diff
@@ -700,6 +700,24 @@ def test_create(self):
     )
 
 
+  def test_multiple_gist_dirs_on_gist_create(self):
+    home_dir = Directory.objects.get_home_directory(self.user)
+
+    gist_dir1 = Directory.objects.create(name=Document2.GIST_DIR, owner=self.user, parent_directory=home_dir)
+    gist_dir2 = Directory.objects.create(name=Document2.GIST_DIR, owner=self.user, parent_directory=home_dir)
+
+    assert_equal(2, Directory.objects.filter(name=Document2.GIST_DIR, type='directory').count())
+
+    response = self._create_gist(
+      statement='SELECT 12345',
+      doc_type='hive-query',
+      name='test_gist_create',
+    )
+
+    assert_equal(1, Directory.objects.filter(name=Document2.GIST_DIR, type='directory').count())
+    assert_equal(Document2.GIST_DIR, Document2.objects.get(type='gist', name='test_gist_create').parent_directory.name)
```

Comment:
Should check for the `id` too, that's the whole point of validating if we deleted the wrong parent dir of the gist :)

Reply:
Very true! Fixing